client/service/src/builder.rs

```diff
@@ -891,7 +877,16 @@ fn build_network<TBl, TExPool, TImpQu, TCl>(
 		system_rpc_rx,
 		has_bootnodes,
 		config.announce_block,
-	).boxed();
+	);
+
+	// The network worker is responsible for gathering all network messages and processing
+	// them. This is quite a heavy task, and at the time of the writing of this comment it
+	// frequently happens that this future takes several seconds or in some situations
+	// even more than a minute until it has processed its entire queue. This is clearly an
+	// issue, and ideally we would like to fix the network future to take as little time as
+	// possible, but we also take the extra harm-prevention measure to execute the networking
+	// future using `spawn_blocking`.
```

Comment:
My _kink_ is super verbose comments about code like these :yum:

Reply:
I could guess that it was written by @tomaka without even checking: https://github.com/paritytech/substrate/commit/f3315cd08963f9c355fdf692526b1fe6ff5ff712